### PR TITLE
Replace the SUI radio button in registration form

### PR DESCRIPTION
**Pull request by foxbunny**
- Implement a RadioButton component to replace the SUI radio button
- Replace usages of SUI radio button within the reigstration form inputs
- Address the issue of SingleChoiceInput needing both fieldset and non-fieldset rendering depending on the settings by modifying the renderAsFieldset option to allow a function that returns the correct value based on the settings

### CODE REVIEW

File: indico/modules/events/registration/client/js/form/FormItem.jsx
Comment (ThiefMaster) [inline indico/modules/events/registration/client/js/form/FormItem.jsx:114]:
```suggestion
    meta,
    ...rest,
```

File: CHANGES.rst
Comment (ThiefMaster) [inline CHANGES.rst:85]:
```suggestion
  keyboard support (:pr:`6621`, thanks :user:`foxbunny`)
```

File: CHANGES.rst
Comment (ThiefMaster) [inline CHANGES.rst:96]: Not sure if this really needs a changelog entry, but I guess someone might be using it in a plugin so no harm in having it ;)

```suggestion
- ``renderAsFieldset`` option in the registration field registry can now be a function that
  returns a boolean (:pr:`6621`, thanks :user:`foxbunny`)
```

File: indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx
Comment (foxbunny) [reply indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx:221]: 1. It should be a toggletip not a tooltip.
2. It's illegal for a label to contain multiple interactive elements for obvious reasons.

Therefore I'll render the toggletip separately, outside the label, and this hack won't be needed.

Comment (foxbunny) [reply indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx:221]: I swear "semantic" is a misnomer. :) Anyway, I'll add it to this PR. If I end up hacking something at least I'll be able to move it to the CSS where it belongs.

**Comment by tomasr8** [comment]
Thanks for the PR!

I don't know if it's just my browser/screen or w/e but the radio does not always look circular:
image
Compare with the current radio:
image

It's also a bit larger now, personally I think I prefer the current smaller size

File: indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx
Comment (foxbunny) [reply indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx:221]: Looks like I'll be doing this in a separate PR after all. I've cleaned up the CSS a bit to get this to work in a more generic way that covers all fieldsets.

Comment (foxbunny) [reply indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx:221]: I'll do a separate PR for the toggletip bit.

**Comment by foxbunny** [comment]
> I don't know if it's just my browser/screen or w/e but the radio does not always look circular:

I wasn't able to reproduce this, but I added `flex: none` to it. Let me know if that fixes it.

> It's also a bit larger now, personally I think I prefer the current smaller size

The idea is that it roughly matches the checkbox size. It's still a bit small as far as recommended click target size goes, and normally I would add some padding around it to compensate, but it would only work if we could make the font sizes a bit larger overall (at least 100% of the default size) which we currently can't. So this is a compromise size.

File: indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx
Comment (ThiefMaster) [reply indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx:221]: OK, but I can't merge this PR until the other fix is ready as well (especially now that the icon itself is also gone)

**Comment by tomasr8** [comment]
> The idea is that it roughly matches the checkbox size. It's still a bit small as far as recommended click target size goes, and normally I would add some padding around it to compensate, but it would only work if we could make the font sizes a bit larger overall (at least 100% of the default size) which we currently can't. So this is a compromise size.

Fair enough :+1: 

> I wasn't able to reproduce this, but I added flex: none to it. Let me know if that fixes it.

It still happens and I don't really know why but it seems to have something to do with fractional widths. `1.2em` ends up being 16.8px which looks off to me. However both 16px and 17px look fine. Here's a comparison between 16.8px and 17px. It's pretty subtle but 16.8px looks like it has sharp corner on the main axes:

image
image(1)

File: indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx
Comment (foxbunny) [reply indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx:221]: What do you mean by "icon is gone"? Can you give me a screenshot?

Comment (ThiefMaster) [reply indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx:221]: `label={<ChoiceLabel choice={c} management={management} paid={isPaidChoice(c)} />}`

You removed this line and replaced it with the plain `label={c.caption}`. The `ChoiceLabel` component is also responsible for showing the warning icon when the field is locked.

Comment (foxbunny) [reply indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx:221]: Hmm, I thought I reverted that, but it looks like the build just crashed. My bad.

**Comment by foxbunny** [comment]
@ThiefMaster Do you want me to do anything else with this PR? I'm sort of needing it for the error messages PR and also to take care of the toggle tip one. The first one is blocked by this, and the second one will be easier to do once this is merged.

**Comment by tomasr8** [comment]
Is there any chance we could avoid fractional sizes like SUI does? I checked on Chrome as well, and the black dot looks a bit off sometimes:

image
image

**Comment by tomasr8** [comment]
Is there any way we could use something like `round()` to get a more consistent look? Otherwise I agree with you that responsiveness is more important, it just feels a bit jarring when compared with the current radio..

**Comment by tomasr8** [comment]
> I haven't heard about any round() that would round to the nearest pixel. In a layout that uses all relative units, that would probably be a bit too unpredictable to be useful.

[round](https://developer.mozilla.org/en-US/docs/Web/CSS/round) looks like it can do it but I just threw it out there as a possibility, I've never used it myself. If you think it's not possible to get around those sub-pixel rendering issues, I trust your judgement :)

**Comment by tomasr8** [comment]
Is there anything you still need to do in this PR? Otherwise, I think it's looking good :)

**Comment by foxbunny** [comment]
No, I'm good

**Review by ThiefMaster — APPROVED** [review]

**Pull request merged**
